app/src/org/commcare/activities/connect/ConnectIdBiometricConfigActivity.java

```diff
@@ -24,22 +31,27 @@ public class ConnectIdBiometricConfigActivity extends CommCareActivity<ConnectId
     private ConnectIdBiometricConfigActivityUiController uiController;
     private BiometricManager biometricManager;
 
+    private BiometricPrompt.AuthenticationCallback biometricPromptCallbacks;
+
+    private boolean attemptingFingerprint = false;
+    private boolean allowPassword = false;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         setTitle(getString(R.string.connect_verify_title));
         biometricManager = BiometricManager.from(this);
-
+        biometricPromptCallbacks = preparePromptCallbacks();
         uiController.setupUI();
-
+        allowPassword = getIntent().hasExtra(ConnectConstants.ALLOW_PASSWORD) && getIntent().getStringExtra(ConnectConstants.ALLOW_PASSWORD).equals("true");;
         BiometricsHelper.ConfigurationStatus fingerprint = BiometricsHelper.checkFingerprintStatus(this,
                 biometricManager);
         BiometricsHelper.ConfigurationStatus pin = BiometricsHelper.checkPinStatus(this, biometricManager);
         if (fingerprint == BiometricsHelper.ConfigurationStatus.NotAvailable &&
                 pin == BiometricsHelper.ConfigurationStatus.NotAvailable) {
             //Skip to password-only workflow
-            finish(true, true);
```

Comment:
This line should stay. This handles the rare case where the user's device doesn't support PIN or fingerprint for screen unlock. When that happens this activity should exit out, and ConnectIdWorkflows then shows a message indicating that the user can not use this device with ConnectID.